mipidsi/src/lib.rs

```diff
@@ -253,4 +253,19 @@ where
         self.dcs
             .write_command(dcs::SetTearingEffect(tearing_effect))
     }
+
+    ///
+    /// Puts the display to sleep, reducing power consumption.
+    /// Need to call [Self::wake] before issuing other commands
+    ///
+    pub fn sleep(&mut self) -> Result<(), Error> {
```

Comment:
Perhaps I'm overcomplicating things but I wonder if we should make this a `self` consuming method that returns some sort of `SleepingDisplay<Display<M...>>` that will only return the `Display<M...>` back by issueing `wake(self) -> Display<M...>` to it.

@rfuest what do you think, perhaps i'm just falling into the Rust overabstraction trap tho.

Reply:
I've dabbled with typestates for other drivers and from what I experienced it becomes a bit of a pain when they are fallible since you need to return the previous state with the error. It's still possible though, just the error case is annoying to handle.

Also not really possible to do async cancellation since it takes ownership if you eventually want the same api for async.

Reply:
Wait, I thought you were proposing a typestate, but a wrapper over it is actually not a bad idea, just annoying for the owner

```rs
pub enum DisplayDriverEnum {
    ASleep(SleepingDisplay<Display<M...>>),
    Awake(Display<M...>)
}
pub struct UI {
   display: DisplayDriverEnum,
}

```

Reply:
> @rfuest what do you think, perhaps i'm just falling into the Rust overabstraction trap tho.

I agree with @xgroleau that adding a typestate of wrapper type will become a hassle for the user of the library and isn't worth it in this case IMO.